modules/cache/jvm/src/main/scala/coursier/cache/internal/Downloader.scala

```diff
@@ -711,6 +728,17 @@ object Downloader {
     helper(sslRetry)
   }
 
+  private object UnknownProtocol {
+    def unapply(t: Throwable): Option[(MalformedURLException, String)] = t match {
+      case ex: MalformedURLException => 
+        Option(ex.getMessage()) match {
+          case Some(msg) if msg.startsWith("unknown protocol: ") => Some((ex, msg))
+          case _ => None
+        }
```

Comment:
`ex.getMessage` shouldn't be too costly to compute, I guess the following should be fine:
```suggestion
      case ex: MalformedURLException if ex.getMessage.startsWith("unknown protocol: ") => 
        Some((ex, ex.getMessage))
```

Reply:
We could even not return the message here, and call `getMessage` when we match `UnknownProtocol` above.